rholang/src/main/k/rholang/tests/Free-Variables/Finding-free-vars-with-logical-or-between-channels.rho

```diff
@@ -0,0 +1,9 @@
+// Logical connectives are slightly nuanced.
+
+// The first match should not execute due to the
+// free variable in the pattern.
+// Since nothing matches, this will rewrite to Nil.
+match for(@{@{x \/ Nil}!(Nil)} <- @Nil){ Nil }
```

Comment:
This won't pass the parser because grammar is as follows:
`_match_ Proc4 "{" [Case] "}"`

and `for( …) { P } ` is Proc2

Reply:
This will:
```
@0!(for(x <- @Nil) { Nil }) | for (@y <- @0) { match y { x => x } }
```

and not only this will parse but it will also match.

Reply:
Thanks for pointing this out; I notice now a slight issue in the way I wrote the framework. I will address this in a future PR, I still need to think about how that affects the framework right now.